api/types.go

```diff
@@ -154,7 +154,7 @@ func (tr Timerange) Slots() int {
 // Timeseries is a single time series, identified with the associated tagset.
 type Timeseries struct {
 	Values []float64
-	Metric TaggedMetric // TODO (jee) - change this to use TagSet instead of TaggedMetric.
+	TagSet TagSet
```

Comment:
this is the change i want to have - everything else is there to support this change.